### PR TITLE
win_chocolatey: name: all state: latest "Error installing all" # fixes:25018

### LINKED ISSUE #25018: win_chocolatey:   name: all     state: latest "Error installing all"
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

win_chocolatey

##### ANSIBLE VERSION

```
$ ansible --version
ansible 2.4.0 (devel 6a3ce18e4b) last updated 2017/05/25 02:48:19 (GMT -500)
  config file =
  configured module search path = [u'/Users/tanner/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/tanner/projects/ansible.git/ansible/lib/ansible
  executable location = /Users/tanner/projects/ansible.git/ansible/bin/ansible
  python version = 2.7.10 (default, Feb  6 2017, 23:53:20) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)]
```

##### CONFIGURATION

ANSIBLE_DEBUG=1

##### OS / ENVIRONMENT

Running Ansible from macOS 10.12.4
Managing Windows7, Windows10

##### SUMMARY

Trying to update all installed chocolatey packages using win_chocolatey and the documented example and it errors out. 

##### STEPS TO REPRODUCE

```yaml
- name: update installed chocolatey packages
  win_chocolatey:
    name: all
    state: latest
```

##### EXPECTED RESULTS

All installed chocolatey packages would get updated

##### ACTUAL RESULTS

```
FAILED! => {
    "changed": false,
    "choco_error_cmd": "choco.exe install -dv -y all -timeout 2700",
    "choco_error_log": null,
    "failed": true,
    "msg": "Error installing all",
    "rc": 1
}
```
choco.summary
```
2017-05-25 03:00:05,817 [INFO ] - ============================================================
2017-05-25 03:00:05,820 [INFO ] - Chocolatey v0.10.5
2017-05-25 03:00:06,035 [WARN ] - Nothing to change. Config already set.
2017-05-25 03:00:08,226 [INFO ] - 0.10.5
2017-05-25 03:00:08,804 [INFO ] - ============================================================
2017-05-25 03:00:08,806 [INFO ] - Chocolatey v0.10.5
2017-05-25 03:00:09,259 [WARN ] - 0 packages installed.
2017-05-25 03:00:09,262 [WARN ] - 
Did you know Pro / Business automatically syncs with Programs and
 Features? Learn more about Package Synchronizer at 
 https://chocolatey.org/compare
2017-05-25 03:00:09,835 [INFO ] - ============================================================
2017-05-25 03:00:09,837 [INFO ] - Chocolatey v0.10.5
2017-05-25 03:00:10,058 [INFO ] - Installing the following packages:
2017-05-25 03:00:10,059 [INFO ] - all
2017-05-25 03:00:10,060 [INFO ] - By installing you accept licenses for the packages.
2017-05-25 03:00:10,589 [ERROR] - all not installed. The package was not found with the source(s) listed.
 If you specified a particular version and are receiving this message, it is possible that the package name exists but the version does not.
 Version: ""
 Source(s): "https://chocolatey.org/api/v2/"
2017-05-25 03:00:10,605 [WARN ] - 
Chocolatey installed 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
2017-05-25 03:00:10,606 [INFO ] - 
2017-05-25 03:00:10,607 [ERROR] - Failures
2017-05-25 03:00:10,608 [ERROR] -  - all - all not installed. The package was not found with the source(s) listed.
 If you specified a particular version and are receiving this message, it is possible that the package name exists but the version does not.
 Version: ""
 Source(s): "https://chocolatey.org/api/v2/"
2017-05-25 03:00:10,610 [WARN ] - 
Enjoy using Chocolatey? Explore more amazing features to take your 
experience to the next level at
 https://chocolatey.org/compare
```

Guessing the documentation is wrong and "all" is no a valid package?

### PR DESCRIPTION
**Pull request by basictheprogram**
##### SUMMARY

Fixes #25018 

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

win_chocolatey

##### ANSIBLE VERSION

```
ansible 2.4.0 (issue-25018 6dd8a4cf78) last updated 2017/05/30 01:14:14 (GMT -500)
  config file =
  configured module search path = [u'/Users/tanner/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/tanner/projects/ansible.git/ansible/lib/ansible
  executable location = /Users/tanner/projects/ansible.git/ansible/bin/ansible
  python version = 2.7.10 (default, Feb  7 2017, 00:08:15) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/modules/windows/win_chocolatey.ps1 [hunk 1]
```diff
@@ -100,6 +100,11 @@ Function Choco-IsInstalled
         [string]$package
     )
 
+   if ($package -like "all")
```
Comment (dagwieers) [inline lib/ansible/modules/windows/win_chocolatey.ps1:103]: Why use `-like`, and not just:
```powershell
if ($package -eq "all") {
    return $true
}
```

**Review by dagwieers — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/windows/win_chocolatey.ps1 (hunk 1 above)
Comment (basictheprogram) [reply lib/ansible/modules/windows/win_chocolatey.ps1:103]: Not a powershell guy, but -like matches "ALL" and "all" and "AlL", etc?

Comment (dagwieers) [reply lib/ansible/modules/windows/win_chocolatey.ps1:103]: And that works with chocolatey ?

Comment (basictheprogram) [reply lib/ansible/modules/windows/win_chocolatey.ps1:103]: I'm not very good at this open source/colab thing. 

Does it normally take 11 days for a trivial patch to get merged into devel? 

If the syntactic sugar needs changing, just change it?

This bug is pretty shallow but waiting 11 days on the fix makes it feel really deep.

Comment (dagwieers) [reply lib/ansible/modules/windows/win_chocolatey.ps1:103]: If you are talking to me specifically, I have no powers.

**Comment by dagwieers** [comment]
@basictheprogram It seems I got that fix from your PR. So this one should have been merged instead of mine. If I had realised today, it would have saved me the effort to make a PR out of it. Sorry for not reading up on IRC about it :-(

File: lib/ansible/modules/windows/win_chocolatey.ps1 (hunk 1 above)
Comment (jborean93) [reply lib/ansible/modules/windows/win_chocolatey.ps1:103]: Sorry for not getting back to this but appreciate the effors @basictheprogram. `-eq` is case insensitive while `-ceq` is the case sensitive version. We merged `-eq` into devel just recently.

**Comment by dagwieers** [comment]
This is implemented by #26417

**Comment by gundalow** [comment]
Closing as it's been fixed by https://github.com/ansible/ansible/pull/26417

**Pull request closed without merging**